src/rpdk/core/contract/suite/handler_commons.py

```diff
@@ -164,7 +164,11 @@ def test_input_equals_output(resource_client, input_model, output_model):
     pruned_output_model = prune_properties_if_not_exist_in_path(
         pruned_output_model, pruned_input_model, resource_client.create_only_paths
     )
+
+    transformed_input_model = resource_client.transform_model(
+        copy.deepcopy(pruned_input_model)
+    )
+
```

Comment:
do you think we need additional copy of it? can we just use transform during the comparison and not as additional input?

Reply:
Hard to use transform during the comparison as pyjq needs inputs with whole model. But updated the function with less inputs and do copy in resource_client.py